applications: serial_lte_modem: Full modem fota support for SLM (#11892)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/FOTA_AT_commands.rst

  * ``0`` - Cancel FOTA (during download only).
  * ``1`` - Start FOTA for application update.
  * ``2`` - Start FOTA for modem delta update.

>   * ``3`` - Start FOTA for full modem update. Activated when overlay-full_fota.conf is used.

Reviewer: ```suggestion
  * ``3`` - Start FOTA for full modem update.
    Activated when the :file:`overlay-full_fota.conf` configuration file is used.
```
"Activated" makes this sound like it happens automatically. Is that the case?

Author: I meant that this feature can be used only when the specific overlay is used. Maybe "Enabled" is better?

Reviewer: Best to keep it simple, and just directly say:
```suggestion
  * ``3`` - Start FOTA for full modem update.
    Can only be used when the :file:`overlay-full_fota.conf` configuration file is used.
```

Author: Fixed

  * ``9`` - Erase delta modem update DFU area.

>   * ``10`` - Start FOTA for full modem update.

Author: Overlay added to the documentation

>   * ``9`` - Erase delta modem update DFU area.

Author: Fixed

>   * ``9`` - Erase delta modem update DFU area.

Reviewer: I don't see these applied.

>   * ``10`` - Start FOTA for full modem update. Activated when overlay-full_fota.conf is used.

Author: Fixed

.. note::

   When doing modem delta FOTA, erasing the modem DFU area is optional since the update process will automatically erase the area if needed.

   However, this leads to the command starting the update taking longer to complete, and also leaves the connection to the FOTA server idle while the area is being erased, which can provoke issues.

.. note::

>    When doing full modem FOTA, an external flash is used to store the firmware. The external flash is ereased automatically after a new firmware activation.

Author: Fixed

>    The activation of a new full modem firmware is done similarly to a delta modem update: even calling XMODEMRESET or resetting the whole device.

Author: Fixed

.. note::

>    When doing modem delta FOTA, erasing the modem DFU area is optional since the update process will automatically erase the area if needed.

Reviewer 2: ```suggestion
   During a delta update to the modem over FOTA, the modem DFU area is automatically erased as needed as part of the update process. Erasing this area manually is optional.
```

Trying to make the structure a little clearer. I hope I understood this correctly.

Author: Fixed

   However, this leads to the command starting the update taking longer to complete, and also leaves the connection to the FOTA server idle while the area is being erased, which can provoke issues.

.. note::

>    When doing full modem FOTA, external flash is used to store the firmware.

Reviewer 2: ```suggestion
   The firmware image is stored in external flash memory during a full modem update over FOTA.
```

Author: Fixed, but the FOTA appreviation has been used.

Reviewer 2: Perhaps 'using FOTA', then.

Author: Fixed accordin to the teams chat

>    The external flash is ereased automatically after a new firmware activation.

Reviewer 2: ```suggestion
   The external flash is erased automatically after a new firmware activation.
```

>    The activation of a new full modem firmware is done identically to a modem delta update, by resetting either the whole device or only the modem.

Reviewer 2: 'Activation' sounds weird to me here, but I can't find a clear precedent on the Cloud side.

My gut says *application* is a better word based on https://docs.nrfcloud.com/Devices/FirmwareUpdate/FOTAOverview.html, but I've also realized we don't ever actually say what this step is called. The entire process is deployment, and in some APIs the act of installing new firmware is called *application*, but that's still more general than what _this_ doc is referring to.

Author: Switching 'activation' and 'application' would be confusing. Here 'activation' means the action to be made and 'application' the software itself which is downloaded.

>    The activation of a new full modem firmware is done identically to a modem delta update, by resetting either the whole device or only the modem.

Reviewer 2: ```suggestion
   Activating the new full modem firmware is done identically to a modem delta update, by resetting either the whole device or only the modem.
```

Author: Fixed

.. note::

   When doing modem delta FOTA, erasing the modem DFU area is optional since the update process will automatically erase the area if needed.

   However, this leads to the command starting the update taking longer to complete, and also leaves the connection to the FOTA server idle while the area is being erased, which can provoke issues.

.. note::

>    When doing full modem FOTA, an external flash is used to store the firmware.
> 
>    The external flash is ereased automatically after a new firmware activation.

Reviewer: ```suggestion
   When doing full modem FOTA, external flash is used to store the firmware.
   The external flash is erased automatically after a new firmware activation.
```

Author: Fixed

> 

Author: Fixed

   However, this leads to the command starting the update taking longer to complete, and also leaves the connection to the FOTA server idle while the area is being erased, which can provoke issues.

   When doing full modem FOTA, an external flash is used to store the firmware. The external flash is ereased automatically after a new firmware activation.

>    The activation of a new full modem firmware is done equally to a delta modem update: evan calling XMODEMRESET or resetting the whole device.

Author: I will wait the documentation review, then I do not have to write it twice ;)

>    The activation of a new full modem firmware is done equally to a delta modem update: evan calling XMODEMRESET or resetting the whole device.

Author: Fixed

.. note::

>    When doing full modem FOTA, an external flash is used to store the firmware.

Reviewer: ```suggestion
   When doing full modem FOTA, external flash is used to store the firmware.
```

Author: Fixed, sorry for several rounds.

Reviewer: No worries. I blame the GitHub preview not displaying the change properly.

   AT#XFOTA=?

>    #XFOTA: (0,1,2,3,6,7,8,9),<file_url>,<sec_tag>,<apn>

Author: Fixed
